### PR TITLE
feat: mysql instrumentation

### PR DESCRIPTION
**Pull request by dyladan**
This contains no functionality changes aside from what it took to do the conversion and pass tests.

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-mysql/package.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    "mysql",
+    "nodejs",
+    "tracing",
+    "profiling",
+    "instrumentation"
+  ],
+  "author": "OpenTelemetry Authors",
+  "license": "Apache-2.0",
+  "engines": {
+    "node": ">=8.0.0"
```
Comment (vmarchaud) [inline plugins/node/opentelemetry-instrumentation-mysql/package.json:31]: `8.5.0`

File: plugins/node/opentelemetry-instrumentation-mysql/package.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  },
+  "devDependencies": {
+    "@opentelemetry/context-async-hooks": "0.18.0",
+    "@opentelemetry/node": "0.18.0",
+    "@opentelemetry/tracing": "0.18.0",
+    "@opentelemetry/test-utils": "0.14.0",
+    "@types/mocha": "7.0.2",
+    "@types/mysql": "2.15.18",
+    "@types/node": "14.0.27",
+    "@types/shimmer": "1.0.1",
```
Comment (vmarchaud) [inline plugins/node/opentelemetry-instrumentation-mysql/package.json:50]: you can remove this one

File: plugins/node/opentelemetry-instrumentation-mysql/package.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    "codecov": "3.7.2",
+    "cross-env": "7.0.2",
```
Comment (vmarchaud) [inline plugins/node/opentelemetry-instrumentation-mysql/package.json:52]: is this needed ?

Comment (dyladan) [reply plugins/node/opentelemetry-instrumentation-mysql/package.json:52]: nope :)

**Review by vmarchaud — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

**Comment by obecny** [comment]
don't merge yet there are more broken links

**Comment by vmarchaud** [comment]
> don't merge yet there are more broken links

You mean in the PR ? I don't see any

I can't rebase your PR @dyladan so you'll need to do it

**Comment by obecny** [comment]
> > don't merge yet there are more broken links
> 
> You mean in the PR ? I don't see any
> 
> Also @dyladan i can't rebase your PR so you'll need to do it

Ignore it, I posted it in wrong PR :/

**Review by obecny — COMMENTED** [review]
can you uncomment mysql in meta package ?

**Comment by dyladan** [comment]
> can you uncomment mysql in meta package ?

done

File: plugins/node/opentelemetry-instrumentation-mysql/src/mysql.ts [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+} from '@opentelemetry/instrumentation';
+import { DatabaseAttribute } from '@opentelemetry/semantic-conventions';
+import type * as mysqlTypes from 'mysql';
+import { MysqlInstrumentationConfig } from './types';
+import { getConnectionAttributes, getDbStatement, getSpanName } from './utils';
+import { VERSION } from './version';
+
+type formatType = typeof mysqlTypes.format;
+
+export class MysqlInstrumentation extends InstrumentationBase<
```
Comment (obecny) [inline plugins/node/opentelemetry-instrumentation-mysql/src/mysql.ts:31]: Previously we had `MySQLPlugin`, the db itself has a name `MySQL` not `Mysql`, it is the same with mongoDB, I think we should stick to name `MySQLInstrumentation`

Comment (dyladan) [reply plugins/node/opentelemetry-instrumentation-mysql/src/mysql.ts]: Done

**Review by obecny — APPROVED** [review]

**Pull request merged**